it/controllers/EnrolCurrentUserControllerISpec.scala

```diff
@@ -0,0 +1,98 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import helpers.IntegrationSpecBase
+import helpers.WiremockHelper._
+import helpers.TestITData._
+import play.api.http.Status
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.controllers.testOnly
+
+class EnrolCurrentUserControllerISpec extends IntegrationSpecBase with Status {
+
+  val teaHost = s"localhost:$port"
+  val urlPath =
+    s"/enrol-pt/enrol-current-user-id"
+
+  s"GET $urlPath" when {
+    "the user is authorised" should {
+      s"return 200 and render the enrol current user page" in {
+        val authResponse = authoriseResponseJson()
+        stubAuthorizePost(OK, authResponse.toString())
+        stubPost(s"/write/.*", OK, """{"x":2}""")
+        val res = buildRequest(urlPath, followRedirects = true)
+          .withHttpHeaders(xSessionId, csrfContent)
+          .get()
+
+        whenReady(res) { resp =>
+          resp.status shouldBe OK
+          resp.body should include(
+            "Which Government Gateway user ID do you want to use to access your personal tax information?"
+          )
+        }
+      }
+    }
+
+    "the user has a session missing required element NINO" should {
+      s"return $UNAUTHORIZED" in {
+        val authResponse = authoriseResponseJson(optNino = None)
+        stubAuthorizePost(OK, authResponse.toString())
+        stubPost(s"/write/.*", OK, """{"x":2}""")
+
+        val res =
+          buildRequest(urlPath).withHttpHeaders(xSessionId, csrfContent).get()
+
+        whenReady(res) { resp =>
+          resp.status shouldBe UNAUTHORIZED
+        }
+      }
+    }
+
+    "the user has a session missing required element Credentials" should {
+      s"return $UNAUTHORIZED" in {
+        val authResponse = authoriseResponseJson(optCreds = None)
+        stubAuthorizePost(OK, authResponse.toString())
+        stubPost(s"/write/.*", OK, """{"x":2}""")
+
+        val res =
+          buildRequest(urlPath).withHttpHeaders(xSessionId, csrfContent).get()
+
+        whenReady(res) { resp =>
+          resp.status shouldBe UNAUTHORIZED
+        }
+      }
+    }
+
+    List(sessionNotFound, insufficientConfidenceLevel).foreach {
+      failureReason =>
+        s"the auth returns 401 and the user has an $failureReason" should {
+          s"return $UNAUTHORIZED" in {
+            stubAuthorizePostUnauthorised(failureReason)
+            stubPost(s"/write/.*", OK, """{"x":2}""")
+
+            val res = buildRequest(urlPath)
+              .withHttpHeaders(xSessionId, csrfContent)
+              .get()
+
+            whenReady(res) { resp =>
+              resp.status shouldBe UNAUTHORIZED
```

Comment:
Unsure about having two tests if they're the exact same thing. If the point is to cover the different Exception types of NoActiveSession and AuthorisationException then this becomes an incorrect test. SessionRecordNotFound should be a 303 status to the login page right?

Reply:
These are not the NoActiveSession. One is no session being present full stop and the other is AuthorisationExceptions for confidence level (we have CL200 rule). The authCode for AuthorisationExceptions returns an unauthorised in both these cases. Also you have this same function for testing is in the LandingPage tests.

Reply:
Is the replicable behaviour? I thought no session being present always returned a NoActiveSession, I haven't ever seen it return an AuthorisationException. Happy to leave this as is, but regardless, but it'd be good to confirm.

Reply:
The error returned is SessionRecordNotFound